### ISSUE TITLE
Trouble when executing puppy-primers

### ISSUE THREAD

**Issue by caya2** [Tropini-lab/PUPpy#15]
Hi! 
I have some problems with the puppy-primers command. First I tried with my cds.fna files and it did not work so I tried with the test files on this page. I still have the same error message... : 

ValueError: Illegal PRIMER_OPT_TM value: 60.0; Illegal PRIMER_MIN_TM value: 58.0; Illegal PRIMER_MAX_TM value: 63.0; Illegal PRIMER_PAIR_MAX_DIFF_TM value: 2.0; Illegal PRIMER_MIN_GC value: 40.0; Illegal PRIMER_MAX_GC value: 60.0

My code is : 
puppy-primers -pr INPUT_primerTarget -i ResultDB.tsv -o ResultPrimers

I use PUPpy with conda.

I tried to change the values that does not work put I have the same error message. I did the update yesterday. Bioinformatics is still new for me so maybe it is an easy fix problem, but I really don't understand what I do wrong.

Thanks for your help !

**Comment by hghezzi** [comment]
Hi!

Thank you for your patience with my delayed response!

Based on your code, I assume that you were able to run puppy-align successfully, managing to create the output folder (default name "alignOUT"). One thing I notice is that you are providing "INPUT_primerTarget" and "ResultDB.tsv" without a specific relative path, which assumes that the folder and file are in the same directory. You can check this by running `ls` in your current working directory. However, I would expect the file "ResultDB.tsv" to require the path of the output folder "./alignOUT/ResultDB.tsv". The new code may look something like this:

`puppy-primers -pr INPUT_primerTarget -i ./alignOUT/ResultDB.tsv -o ResultPrimers`

Please let me know if this is not the problem and we can try to work on a new solution!

TLDR: Make sure your filepaths are correct :)

**Comment by LucoDevro** [comment]
Hi!

I'm having a similar issue. I downloaded a couple of genomic CDS files from NCBI, added `cds` to the filenames, and distributed them in a target and a non-target folder. I successfully ran `puppy-align`, which put its output (`ResultDB.tsv`) in a separate `align` subfolder, using this command:

`puppy-align -pr target/ -nt non_target/ -o align/`

However, when I ran `puppy-primers` using the command below, it fails with similar illegal value errors as above in this thread.

`puppy-primers -pr target/ -i ./align/ResultDB.tsv -o primer/ -ng 1 -np 1 -mips 18 -maps 30 -migc 40 -magc 60 -tmd 5 -s 200-5000`

Do these Illegal value errors mean I'm putting the primer specs too stringent, or is it a code issue? As you see, I already used some more relaxed values that still work well from our experience in the lab.

The directory structure I run both commands in looks like this:

`ls: align  non_target  primer  target`

**Comment by caya2** [comment]
Hi,
thanks for your answer ! 
Puppy-align did work. 
The files where in the good directory, since I wasn't sure if the path was the problem I moved the ResultDB.tsv document in the folder I was in. So the problem is not the path... When I tried to do groups primer with these same files it worked. But specific primers doesn't... 

I don't know if you have an other solution...

Thanks!

**Comment by mr2raccoon** [comment]
Hey, I ran into a similar problem when running `puppy-primers`. Did anyone find a solution?

I first ran it with standard settings received a similar error and then relaxed the settings to : ```-mitm 50.0 -matm 67.0 -magc 65.0``` and still received a similar error.

```  File "primer3/thermoanalysis.pyx", line 1273, in primer3.thermoanalysis._ThermoAnalysis.run_design
  File "primer3/thermoanalysis.pyx", line 1201, in primer3.thermoanalysis._ThermoAnalysis._set_globals_and_seq_args
  File "primer3/thermoanalysis.pyx", line 1182, in primer3.thermoanalysis._ThermoAnalysis._set_globals_and_seq_args
  File "primer3/thermoanalysis.pyx", line 1437, in primer3.thermoanalysis.pdh_wrap_set_seq_args_globals
ValueError: Illegal PRIMER_OPT_TM value: 60.0; Illegal PRIMER_MIN_TM value: 50.0; Illegal PRIMER_MAX_TM value: 67.0; Illegal PRIMER_PAIR_MAX_DIFF_TM value: 2.0; Illegal PRIMER_MIN_GC value: 40.0; Illegal PRIMER_MAX_GC value: 65.0
```

```puppy-align``` seemed to have worked. I get UniqueGenesList.tsv with 185 unique genes. (annotation columns says no matching annotation for each). Strangely, the UniqueGenesPlot.pdf shows unique Genes = 0 ? 
FYI: My target was 1 strain of a species. The non-targets were the `cds.fna-files` of all other strains of this specific species. Maybe to little variability?


Thanks for creating the tool and any help!

**Comment by Hiumin** [comment]
I don't know if this information will be helpful to anyone, but `puppy-primers` was running fine for me until I accidentally used a different `micromamba` instance (on my university server) to activate the environment. That was when I got this exact same error. Trying out different values for the various "illegal" parameters did not help. The problem persisted with and without non-target CDSs. In the end, everything was fine again after I switched back to the usual `micromamba` version, but I still don't know why it happened.
